### PR TITLE
[Aparat] Add Multiple Formats Support

### PR DESCRIPTION
**Pull request by mxamin**
## Please follow the guide below
- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your _pull request_ (like that [x])
- Use _Preview_ tab to see how your _pull request_ will actually look like

---
### Before submitting a _pull request_ make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
### What is the purpose of your _pull request_?
- [ ] Bug fix
- [ ] New extractor
- [x] New feature

---
### Description of your _pull request_ and other information

the old version of `Aparat` extractor only downloads the lowest quality video, but the new version (this pull request) supports downloading videos with different qualities, from lowest to highest (`formats` keyword).

### CODE REVIEW

File: youtube_dl/extractor/aparat.py [hunk 1]
```diff
@@ -32,27 +29,70 @@ def _real_extract(self, url):
...
+        )
+        file_list_pseudo = self._parse_json(
+            self._search_regex(
+                r'fileListPseudo\s*=\s*JSON\.parse\(\'([^\']+)\'\)',
+                webpage,
+                'file list pseudo',
+                default='[]'
+            ),
+            video_id
+        )
```
Comment (dstftw) [inline youtube_dl/extractor/aparat.py]: Eliminate code duplication.

File: youtube_dl/extractor/aparat.py [hunk 2]
```diff
@@ -32,27 +29,70 @@ def _real_extract(self, url):
...
+
+        total_file_list = []
+        if file_list:
+            total_file_list.extend(file_list[0])
+
+        if file_list_pseudo:
+            total_file_list.extend(file_list_pseudo[0])
```
Comment (dstftw) [inline youtube_dl/extractor/aparat.py]: Eliminate code duplication.

File: youtube_dl/extractor/aparat.py [hunk 3]
```diff
@@ -32,27 +29,70 @@ def _real_extract(self, url):
...
+            video = {}
+            video['url'] = item['file']
+            video['format'] = item['type']
+            video['ext'] = 'mp4'
+            video_label = item.get('label', 'unknown')
+            video['label'] = labels.get(video_label, 0)
+
+            formats.append(video)
+
+        formats = sorted(formats, key=lambda x: x['label'])
```
Comment (dstftw) [inline youtube_dl/extractor/aparat.py]: Use `_sort_formats` instead.

File: youtube_dl/extractor/aparat.py [hunk 4]
```diff
@@ -32,27 +29,70 @@ def _real_extract(self, url):
...
 
-        file_list = self._parse_json(self._search_regex(
-            r'fileList\s*=\s*JSON\.parse\(\'([^\']+)\'\)', webpage, 'file list'), video_id)
-        for i, item in enumerate(file_list[0]):
-            video_url = item['file']
-            req = HEADRequest(video_url)
-            res = self._request_webpage(
-                req, video_id, note='Testing video URL %d' % i, errnote=False)
-            if res:
-                break
```
Comment (dstftw) [inline youtube_dl/extractor/aparat.py]: Why is this check removed?

Comment (mxamin) [inline youtube_dl/extractor/aparat.py]: I thought there is no need for video url checking, any how I put it back

File: youtube_dl/extractor/aparat.py (hunk 1 above)
Comment (mxamin) [inline youtube_dl/extractor/aparat.py]: done

File: youtube_dl/extractor/aparat.py (hunk 2 above)
Comment (mxamin) [inline youtube_dl/extractor/aparat.py]: done

File: youtube_dl/extractor/aparat.py (hunk 3 above)
Comment (mxamin) [inline youtube_dl/extractor/aparat.py]: done

**Comment by mxamin** [comment]
I have done the requested changes, is there anything left to do?

**Pull request closed without merging**